### ISSUE TITLE
Could not initialize class AppEngineStandardFacet after install 

### ISSUE THREAD

**Issue by gssolimano** [GoogleCloudPlatform/google-cloud-eclipse#3623]
- Cloud Tools for Eclipse version: 1.8.3
- Google Cloud SDK version: 304.0.0
- Eclipse version: 2020-06 (4.16.0)
- OS: Windows 10.0.18363 Build 18363
- Java version: 1.8.0_261

I installed google cloud tools
After restart eclipse a get error
 
"_An error has occurred. See error log for more details.
Could not initialize class 
com.google.cloud.tools.eclipse.appengine.facets.AppEngineStandardFacet_"

![image](https://user-images.githubusercontent.com/29986657/90174298-24b8de80-dd7c-11ea-99c4-6877061140f7.png)

**Comment by briandealwis** [comment]
Hi @gssolimano — is it possible to include your `<workspace>/.metadata/.log` file?

**Comment by gssolimano** [comment]
Yes off course 

[metadata.log](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/files/5071441/metadata.log)

**Comment by briandealwis** [comment]
Hmm, very early on you're getting an odd failure that 

    Version JRE8 of project facet com.google.cloud.tools.eclipse.appengine.facets.standard does not exist.

which would seem to indicate that there was some problem loading the `com.google.cloud.tools.eclipse.appengine.standard.java8` bundle.

Could you try the following:

  1. Open the _Console_ view, click on the new console button, and select _Host OSGi Console_:
       <img width="297" alt="Screen Shot 2020-08-14 at 12 32 48 AM" src="https://user-images.githubusercontent.com/202851/90214248-dc261300-ddc5-11ea-90fd-b196d2b1cb68.png">
  2. At the `osgi>` prompt, type `diag com.google.cloud.tools.eclipse.appengine.standard.java8`.

When I try this with 2020-06, I see:

    osgi> diag com.google.cloud.tools.eclipse.appengine.standard.java8
    com.google.cloud.tools.eclipse.appengine.standard.java8 [814]
      No resolution report for the bundle.
    osgi> 

<details>
<summary>the relevant exception</summary>

```
!ENTRY org.eclipse.jface 4 2 2020-08-13 15:44:37.827
!MESSAGE Problems occurred when invoking code from plug-in: "org.eclipse.jface".
!STACK 0
java.lang.ExceptionInInitializerError
	at com.google.cloud.tools.eclipse.appengine.facets.ui.navigator.AppEngineContentProvider.isAppEngine(AppEngineContentProvider.java:99)
	at com.google.cloud.tools.eclipse.appengine.facets.ui.navigator.AppEngineLabelProvider.getStyledText(AppEngineLabelProvider.java:56)
[...]
Caused by: java.lang.IllegalArgumentException: Version JRE8 of project facet com.google.cloud.tools.eclipse.appengine.facets.standard does not exist.
	at org.eclipse.wst.common.project.facet.core.util.internal.Versionable.getVersion(Versionable.java:80)
	at org.eclipse.wst.common.project.facet.core.internal.ProjectFacet.getVersion(ProjectFacet.java:1)
	at com.google.cloud.tools.eclipse.appengine.facets.AppEngineStandardFacet.<clinit>(AppEngineStandardFacet.java:59)
	... 136 more
```


</details>

**Comment by gssolimano** [comment]
Hi @briandealwis, thanks for your help. 

I ron the command, but i get a diferent resuelt: 

```
osgi> diag com.google.cloud.tools.eclipse.appengine.standard.java8
com.google.cloud.tools.eclipse.appengine.standard.java8 [1056]
  Unresolved requirement: Require-Bundle: com.google.cloud.tools.eclipse.appengine.localserver; bundle-version="0.1.0"
    -> Bundle-SymbolicName: com.google.cloud.tools.eclipse.appengine.localserver; bundle-version="0.1.0.201910291000"; singleton:="true"
       com.google.cloud.tools.eclipse.appengine.localserver [1054]
         Unresolved requirement: Import-Package: com.google.cloud.tools.eclipse.sdk.ui.preferences
           -> Export-Package: com.google.cloud.tools.eclipse.sdk.ui.preferences; bundle-symbolic-name="com.google.cloud.tools.eclipse.sdk.ui"; bundle-version="0.1.0.201910291000"; version="0.0.0"
              com.google.cloud.tools.eclipse.sdk.ui [1068]
                Unresolved requirement: Import-Package: org.eclipse.mylyn.commons.ui.dialogs
                  -> Export-Package: org.eclipse.mylyn.commons.ui.dialogs; bundle-symbolic-name="org.eclipse.mylyn.commons.ui"; bundle-version="3.25.1.v20200511-1958"; version="0.0.0"
                     org.eclipse.mylyn.commons.ui [626]
                       Bundle was not resolved because of a uses constraint violation.
  org.apache.felix.resolver.reason.ReasonException: Uses constraint violation. Unable to resolve resource org.eclipse.mylyn.commons.ui [osgi.identity; type="osgi.bundle"; version:Version="3.25.1.v20200511-1958"; osgi.identity="org.eclipse.mylyn.commons.ui"; singleton:="true"] because it is exposed to package 'javax.annotation' from resources org.eclipse.osgi [osgi.identity; type="osgi.bundle"; version:Version="3.15.300.v20200520-1959"; osgi.identity="org.eclipse.osgi"; singleton:="true"] and javax.annotation [osgi.identity; type="osgi.bundle"; version:Version="1.3.5.v20200504-1837"; osgi.identity="javax.annotation"] via two dependency chains.

Chain 1:
  org.eclipse.mylyn.commons.ui [osgi.identity; type="osgi.bundle"; version:Version="3.25.1.v20200511-1958"; osgi.identity="org.eclipse.mylyn.commons.ui"; singleton:="true"]
    require: (osgi.wiring.bundle=org.eclipse.core.runtime)
     |
    provide: osgi.wiring.bundle: org.eclipse.core.runtime
  org.eclipse.osgi [osgi.identity; type="osgi.bundle"; version:Version="3.15.300.v20200520-1959"; osgi.identity="org.eclipse.osgi"; singleton:="true"]

Chain 2:
  org.eclipse.mylyn.commons.ui [osgi.identity; type="osgi.bundle"; version:Version="3.25.1.v20200511-1958"; osgi.identity="org.eclipse.mylyn.commons.ui"; singleton:="true"]
    require: (&(osgi.wiring.bundle=com.google.guava)(&(bundle-version>=21.0.0)(!(bundle-version>=22.0.0))))
     |
    provide: osgi.wiring.bundle; bundle-version:Version="21.0.0.v20170206-1425"; osgi.wiring.bundle="com.google.guava"
  com.google.guava [osgi.identity; type="osgi.bundle"; version:Version="21.0.0.v20170206-1425"; osgi.identity="com.google.guava"]
    import: (osgi.wiring.package=javax.annotation)
     |
    export: osgi.wiring.package: javax.annotation
  javax.annotation [osgi.identity; type="osgi.bundle"; version:Version="1.3.5.v20200504-1837"; osgi.identity="javax.annotation"]                Bundle was not resolved because of a uses constraint violation.
  org.apache.felix.resolver.reason.ReasonException: Uses constraint violation. Unable to resolve resource org.eclipse.mylyn.commons.ui [osgi.identity; type="osgi.bundle"; version:Version="3.25.1.v20200511-1958"; osgi.identity="org.eclipse.mylyn.commons.ui"; singleton:="true"] because it is exposed to package 'javax.annotation' from resources org.eclipse.osgi [osgi.identity; type="osgi.bundle"; version:Version="3.15.300.v20200520-1959"; osgi.identity="org.eclipse.osgi"; singleton:="true"] and javax.annotation [osgi.identity; type="osgi.bundle"; version:Version="1.3.5.v20200504-1837"; osgi.identity="javax.annotation"] via two dependency chains.

Chain 1:
  org.eclipse.mylyn.commons.ui [osgi.identity; type="osgi.bundle"; version:Version="3.25.1.v20200511-1958"; osgi.identity="org.eclipse.mylyn.commons.ui"; singleton:="true"]
    require: (osgi.wiring.bundle=org.eclipse.core.runtime)
     |
    provide: osgi.wiring.bundle: org.eclipse.core.runtime
  org.eclipse.osgi [osgi.identity; type="osgi.bundle"; version:Version="3.15.300.v20200520-1959"; osgi.identity="org.eclipse.osgi"; singleton:="true"]

Chain 2:
  org.eclipse.mylyn.commons.ui [osgi.identity; type="osgi.bundle"; version:Version="3.25.1.v20200511-1958"; osgi.identity="org.eclipse.mylyn.commons.ui"; singleton:="true"]
    require: (&(osgi.wiring.bundle=com.google.guava)(&(bundle-version>=21.0.0)(!(bundle-version>=22.0.0))))
     |
    provide: osgi.wiring.bundle; bundle-version:Version="21.0.0.v20170206-1425"; osgi.wiring.bundle="com.google.guava"
  com.google.guava [osgi.identity; type="osgi.bundle"; version:Version="21.0.0.v20170206-1425"; osgi.identity="com.google.guava"]
    import: (osgi.wiring.package=javax.annotation)
     |
    export: osgi.wiring.package: javax.annotation
  javax.annotation [osgi.identity; type="osgi.bundle"; version:Version="1.3.5.v20200504-1837"; osgi.identity="javax.annotation"]         Bundle was not resolved because of a uses constraint violation.
  org.apache.felix.resolver.reason.ReasonException: Uses constraint violation. Unable to resolve resource org.eclipse.mylyn.commons.ui [osgi.identity; type="osgi.bundle"; version:Version="3.25.1.v20200511-1958"; osgi.identity="org.eclipse.mylyn.commons.ui"; singleton:="true"] because it is exposed to package 'javax.annotation' from resources org.eclipse.osgi [osgi.identity; type="osgi.bundle"; version:Version="3.15.300.v20200520-1959"; osgi.identity="org.eclipse.osgi"; singleton:="true"] and javax.annotation [osgi.identity; type="osgi.bundle"; version:Version="1.3.5.v20200504-1837"; osgi.identity="javax.annotation"] via two dependency chains.

Chain 1:
  org.eclipse.mylyn.commons.ui [osgi.identity; type="osgi.bundle"; version:Version="3.25.1.v20200511-1958"; osgi.identity="org.eclipse.mylyn.commons.ui"; singleton:="true"]
    require: (osgi.wiring.bundle=org.eclipse.core.runtime)
     |
    provide: osgi.wiring.bundle: org.eclipse.core.runtime
  org.eclipse.osgi [osgi.identity; type="osgi.bundle"; version:Version="3.15.300.v20200520-1959"; osgi.identity="org.eclipse.osgi"; singleton:="true"]

Chain 2:
  org.eclipse.mylyn.commons.ui [osgi.identity; type="osgi.bundle"; version:Version="3.25.1.v20200511-1958"; osgi.identity="org.eclipse.mylyn.commons.ui"; singleton:="true"]
    require: (&(osgi.wiring.bundle=com.google.guava)(&(bundle-version>=21.0.0)(!(bundle-version>=22.0.0))))
     |
    provide: osgi.wiring.bundle; bundle-version:Version="21.0.0.v20170206-1425"; osgi.wiring.bundle="com.google.guava"
  com.google.guava [osgi.identity; type="osgi.bundle"; version:Version="21.0.0.v20170206-1425"; osgi.identity="com.google.guava"]
    import: (osgi.wiring.package=javax.annotation)
     |
    export: osgi.wiring.package: javax.annotation
  javax.annotation [osgi.identity; type="osgi.bundle"; version:Version="1.3.5.v20200504-1837"; osgi.identity="javax.annotation"]  Bundle was not resolved because of a uses constraint violation.
  org.apache.felix.resolver.reason.ReasonException: Uses constraint violation. Unable to resolve resource org.eclipse.mylyn.commons.ui [osgi.identity; type="osgi.bundle"; version:Version="3.25.1.v20200511-1958"; osgi.identity="org.eclipse.mylyn.commons.ui"; singleton:="true"] because it is exposed to package 'javax.annotation' from resources org.eclipse.osgi [osgi.identity; type="osgi.bundle"; version:Version="3.15.300.v20200520-1959"; osgi.identity="org.eclipse.osgi"; singleton:="true"] and javax.annotation [osgi.identity; type="osgi.bundle"; version:Version="1.3.5.v20200504-1837"; osgi.identity="javax.annotation"] via two dependency chains.

Chain 1:
  org.eclipse.mylyn.commons.ui [osgi.identity; type="osgi.bundle"; version:Version="3.25.1.v20200511-1958"; osgi.identity="org.eclipse.mylyn.commons.ui"; singleton:="true"]
    require: (osgi.wiring.bundle=org.eclipse.core.runtime)
     |
    provide: osgi.wiring.bundle: org.eclipse.core.runtime
  org.eclipse.osgi [osgi.identity; type="osgi.bundle"; version:Version="3.15.300.v20200520-1959"; osgi.identity="org.eclipse.osgi"; singleton:="true"]

Chain 2:
  org.eclipse.mylyn.commons.ui [osgi.identity; type="osgi.bundle"; version:Version="3.25.1.v20200511-1958"; osgi.identity="org.eclipse.mylyn.commons.ui"; singleton:="true"]
    require: (&(osgi.wiring.bundle=com.google.guava)(&(bundle-version>=21.0.0)(!(bundle-version>=22.0.0))))
     |
    provide: osgi.wiring.bundle; bundle-version:Version="21.0.0.v20170206-1425"; osgi.wiring.bundle="com.google.guava"
  com.google.guava [osgi.identity; type="osgi.bundle"; version:Version="21.0.0.v20170206-1425"; osgi.identity="com.google.guava"]
    import: (osgi.wiring.package=javax.annotation)
     |
    export: osgi.wiring.package: javax.annotation
  javax.annotation [osgi.identity; type="osgi.bundle"; version:Version="1.3.5.v20200504-1837"; osgi.identity="javax.annotation"]
```

**Comment by briandealwis** [comment]
tl;dr: Please try running under Java 11.

Ultimately the problem here is that you have installed something that is pulling in a bundle called `javax.annotation v1.3.5.v20200504-1837` which is providing clashing variants of classes that are provided by your Java 8 JRE.  These classes `javax.annotation` classes were removed in Java 11 and so this shouldn't be a problem in Java 11.

So here's what's happening:
  - JavaEE defined a set of annotation classes in `javax.annotation`, including the very useful `@PostConstruct` and `@PreDestroy` annotations used by dependency injection systems, including Eclipse's own.
  - The `javax.annotation.**` classes weren't included in Java 7, and so Eclipse Orbit introduced a `javax.annotation` bundle that included these annotation classses.
  - Java 8 then included several `javax.annotation.*` classes in its runtime.  You can see this with _Open Type_: note how there is a `javax.annotation.PreDestroy` class in the _JavaSE 8_ (my JRE) as well as in the `javax.annotation 1.2.0` bundle.
    ![Screen Shot 2020-08-14 at 10 33 56 AM](https://user-images.githubusercontent.com/202851/90262322-2a660100-de1c-11ea-8cfb-dd89aad7861c.png)
  - The `javax.annotation` bundle from Eclipse Orbit was changed to [both import and export its `javax.annotation.**` packages](https://bugs.eclipse.org/bugs/show_bug.cgi?id=348155#c8).  This [importing-and-exporting of packages](https://blog.osgi.org/2007/04/importance-of-exporting-nd-importing.html) ensures that the bundle will use and export the `javax.annotation.**` classes from the JRE (via the _system bundle_, the OSGi framework, which is called `org.eclipse.osgi` in Eclipse Equinox) _in preference of its own classes when available_.
  - As part of the modularization effort with Java 9, the `javax.annotation.**` classes were removed from the `java.base` and were instead introduced as separate modules.  JavaEE published their own `javax.annotation` module.  JavaEE was donated to the Eclipse Foundation and renamed JakartaEE.
  - On August 2, 2019, the JakartaEE project [published a release](https://search.maven.org/artifact/jakarta.annotation/jakarta.annotation-api/1.3.5/jar) of these [common annotations versioned 1.3.5](https://github.com/eclipse-ee4j/common-annotations-api/tree/1.3.5/api) where the classes were still using the `javax.annotation` prefix.  **Unfortunately this bundle does not use the import and export convention.**
     - This is no longer a problem as the JakartaEE project is renaming its packages from javax.* to jakarta
  - This release was then imported into [Eclipse Orbit](https://git.eclipse.org/r/c/orbit/orbit-recipes/+/161789) as part of an effort to [replace the no-longer maintained Spotify Docker Client](https://bugs.eclipse.org/bugs/show_bug.cgi?id=558284), and this import+export was not caught.

We're getting caught in the middle of this as both `org.eclipse.mylyn.commons.ui` and `com.google.guava` import the `javax.annotation` package and are getting conflicting providers.

**Comment by briandealwis** [comment]
You can [find out which features are pulling in the `javax.annotation` bundle](https://stackoverflow.com/a/11014577/600339) with:
```
osgi> start org.eclipse.equinox.p2.console
osgi> provlpquery this "select(parent | parent.properties['org.eclipse.equinox.p2.type.group'] == true && parent.requirements.exists(rc | everything.exists(iu | iu.id == 'javax.annotation' && iu ~= rc)))" true
org.eclipse.e4.rcp.feature.group 4.16.0.v20200604-0951
```

**Comment by briandealwis** [comment]
Filed [Eclipse Orbit bug 566085](https://bugs.eclipse.org/bugs/show_bug.cgi?id=566085).

**Comment by juergen-albert** [comment]
This might give you some help to Analyse your issue: https://blogs.paremus.com/2011/09/solving-osgi-uses-constraint-violations/ The described issue looks spot on to me!

**Comment by maho7791** [comment]
I finally found two different problems here:

1. Multiple chains resolving javax.annotation.*
2. Split package problem with javax.annotation.* and JSR 305

1. Multiple chains
This problem is related to the usage of require bundle instead of import package in org.eclipse.mylyn.commons.ui.
First the require-bundle org.eclipse.core.runtime in mylyn.commons.ui forces the possible wire of javax.annotation from the osgi framework, resp. the JVM over org.eclipse.osgi. This goes back to the require-bundle org.eclipse.osgi declaration in org.eclipse.core.runtime.
The second chain goes over the require-bundle com.google.guava [21;22) declaration in mylyn.commons.ui. Guava 21 then has an import package for javax.annotation without version means [0.0.0;infinte). This is fullfilled by the javax.annotation bundle 1.3.5 because of its exports.

Because of these two require bundles declaration in mylyn.commons.ui, there will be no valid bundle space for javax.annotation.
Thats it.
In this case the problem is guava 21, with its unversioned import of javax.annotation. But this is fixed since version 25. 
Using this version is not possible with mylyn, because they use the version range restriction in require bundle.

The point 2 is, that the import javax.annotation in guava is not meant for the Java Annotation API but rather to JS305! This is a split package problem. Guava tries to optionally resolve against this dependency:

https://repo.maven.apache.org/maven2/com/google/code/findbugs/jsr305/3.0.2/

This is JSR305 - https://jcp.org/en/jsr/detail?id=305. This is **not** the same like the Java Annotations API that now comes from Jakarta. 
https://repo.maven.apache.org/maven2/jakarta/annotation/jakarta.annotation-api/

Like described above newer versions of Guava (since 25) use an versioned import package. That import should always match their findbug-jsr305 and not the one from the jakarta or the orbit. 
For this case it would mean, that guave would resolve, also with no optional dependency for package javax.annotation in version range [3.0,4). The javax.annotation 1.3.5 bundle wouldn't be a candidate for its import and also the framework export in version 0.0.0 would be no candiate. In the end there would be only one chain.

I think this is an example, why not to use require-bundle, because it causes more problems, then the convenience brings.

**Comment by diego92sigma6** [comment]
Closing issue due to cleanup. Please reopen if needed
